strength_percentiles.py

```diff
@@ -176,12 +199,189 @@ def populate_database(webpage, database):
+        'bench', or 'deadlift'
+    lift: string
+        Either 'squat', 'bench', or 'deadlift'.
+    Returns:
+    --------
+    mean_lift: float
+        The average of every lift (either squat, bench, or deadlift),
+        entered in the dataframe.
+
+    """
+    means = dataframe.mean()
+    mean_lift = means[lift]
+    return mean_lift
+
+
+def get_categories_from_user():
+    """
+    Prompts users for information on their gender, professional status, and
+    equipment category that they compete in. Returns a dictionary containing
+    the user's responses.
+    Returns:
+    --------
+    user_categories: dictionary
+        A dictionary with category names ('gender', 'professional_status'
+        and 'equipment') as keys and the users responses as values.
+
+    """
+    print "Please respond to the following prompts. "
+    gender = 'gender'
+    professional_status = 'professional_status'
+    equipment = 'equipment'
+    user_categories = {GENDER: None, PROFESSIONAL_STATUS: None, EQUIPMENT: None}
+    for category_name, accepted_values in CATEGORY_VALUES.iteritems():
+        prompt_for_category = "What is your %s? Please enter one of %r, " \
+            % (category_name, accepted_values) + " or press Return to be " + \
+            "compared to both."
+        while True:
+            response = raw_input(prompt_for_category)
+            # User should either enter the category they want to be compared
+            # against, or an empty string to be compared to both.  If their
+            # response was not one of these, prompt them to try again.
+            if response not in accepted_values and response:
+                print "Your entry was not valid.  Please try again."
+                continue
+            user_categories[category_name] = response
+            break
+    return user_categories
+
+
+def get_lifts_from_user():
+    """
+    Prompts users for information on their lifts.  Returns a dictionary
+    containing the user's responses.
+    Returns:
+    --------
+    lifts: dictionary
+        A dictionary with lift names ('squat', 'bench', 'deadlift') and
+        'total' as keys and the users responses as values, with the exception
+        of lifts['total'].  The lifters total is calculated and loaded into
+        the dictionary if valid numeric values were given for all three lifts,
+        and None is loaded for the total if there are missing values.
+
+    """
+    print "Please enter your lifts below in lbs."
+    lifts = {SQUAT: None, BENCH: None, DEADLIFT: None}
+    for lift in lifts:
+        prompt_for_lift = "What is your %s? Hit return to skip this lift." \
+            % lift
+        while True:
+            try:
+                response = raw_input(prompt_for_lift)
+                if not response: # user entered empty string
+                    response = None
+                    break
+                response = float(response)
```

Comment:
You might need to do some proper user input handling. What if the user enters non-numeric gibberish? Might need a try/except block. Might be good to offload that logic into it's own function if the try/except block gets too big.